pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -56,6 +57,22 @@ func (a *Adapter) getPod(refresh bool) (*corev1.Pod, error) {
 	if refresh || a.pod == nil {
 		podSelector := fmt.Sprintf("component=%s", a.ComponentName)
 
+		// First check if the pod is already in desired state
+		c := a.Client.GetKubeClient()
+		podList, err := c.KubeClient.CoreV1().Pods(c.Namespace).List(context.TODO(), metav1.ListOptions{
```

Comment:
I think we can avoid listing the pods and use the condition `appsv1.DeploymentAvailable` in the `WaitAndGetPodWithEvents` function to start and stop the spinner

Reply:
In fact, I cannot see any use case where this function getPod is really waiting for the pod to be started. @mik-dass do you have any example when this function is waiting?

Reply:
I changed to start the spinner only when a phase different from the expected one is received